sycl/source/detail/pi.cpp

```diff
@@ -142,10 +142,9 @@ std::string platformInfoToString(pi_platform_info info) {
     return "PI_PLATFORM_INFO_VENDOR";
   case PI_PLATFORM_INFO_EXTENSIONS:
     return "PI_PLATFORM_INFO_EXTENSIONS";
-  default:
-    die("Unknown pi_platform_info value passed to "
-        "cl::sycl::detail::pi::platformInfoToString");
   }
+  die("Unknown pi_platform_info value passed to "
```

Comment:
NIT: This is strange that compiler didn't see this unreachable (if I correctly understand what warning you are trying to fix). But, don't we have a coding standard that each "switch" should have a "default"?

Reply:
I believe this fix here is for -Wcovered-switch-default warning, i.e. when all cases in switch are listed, but the default statement is still there (i.e. redundant).

Reply:
OK, here it is safe to dies after the switch.